rust/pattern/constraint/type_/owns.rs

```diff
@@ -20,54 +20,44 @@
  *
  */
 
-use std::{fmt, iter};
+use std::{collections::HashSet, fmt, iter};
 
 use crate::{
     common::{error::collect_err, token, validatable::Validatable, Result},
     pattern::{variable::Reference, Type, TypeVariable, TypeVariableBuilder, UnboundVariable},
     Label,
 };
 
-#[derive(Debug, Clone, Copy, Eq, PartialEq)]
-pub enum IsKeyAttribute {
-    Yes,
-    No,
+#[derive(Debug, Clone, Copy, Eq, Hash, PartialEq)]
+pub enum Annotation {
+    Key,
+    Unique,
 }
 
-impl From<bool> for IsKeyAttribute {
-    fn from(is_key: bool) -> Self {
-        match is_key {
-            true => IsKeyAttribute::Yes,
-            false => IsKeyAttribute::No,
-        }
-    }
-}
-
-pub const KEY: IsKeyAttribute = IsKeyAttribute::Yes;
-
-impl fmt::Display for IsKeyAttribute {
+impl fmt::Display for Annotation {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if *self == IsKeyAttribute::Yes {
-            write!(f, " {}", token::Constraint::IsKey)?;
+        write!(f, "@")?;
+        match self {
+            Self::Key => write!(f, "{}", token::Annotation::Key),
+            Self::Unique => write!(f, "{}", token::Annotation::Unique),
         }
-        Ok(())
     }
 }
 
 #[derive(Debug, Clone, Eq, PartialEq)]
 pub struct OwnsConstraint {
     pub attribute_type: TypeVariable,
     pub overridden_attribute_type: Option<TypeVariable>,
-    pub is_key: IsKeyAttribute,
+    pub annotations: HashSet<Annotation>,
```

Comment:
Potentially could be a bitset instead, which may or may not save a bit of overhead.